src/org/usfirst/frc/team2335/robot/OperatorInterface.java

```diff
@@ -1,52 +1,99 @@
 
+import org.usfirst.frc.team2335.robot.commands.AutoFromStation1;
 import org.usfirst.frc.team2335.robot.commands.CenterRobot;
+import org.usfirst.frc.team2335.robot.commands.IntakeBalls;
+import org.usfirst.frc.team2335.robot.commands.Lift;
 import org.usfirst.frc.team2335.robot.commands.PositionRobot;
+import org.usfirst.frc.team2335.robot.commands.ShootGroup;
 import org.usfirst.frc.team2335.robot.commands.Strafe;
 import org.usfirst.frc.team2335.robot.triggers.Axis;
 
 import edu.wpi.first.wpilibj.Joystick;
 import edu.wpi.first.wpilibj.buttons.JoystickButton;
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 
-//TODO: Rename to OperatorInterface after basic functionality for clarity.
-//This class connects operator control to different commands
-public class OI
+public class OperatorInterface
 {	
 	Joystick controller, xbox;
+	
+	//Auto test buttons
+	JoystickButton autoTest;
+	
+	//Climb buttons
+	JoystickButton climb;
+	
+	//Shoot buttons
+	JoystickButton shootBall, intakeBalls;
+	
+	//Vision buttons
 	JoystickButton center, position;
 	Axis strafe;
 	
 	/**
 	 * OperatorInterface constructor. Always initialize last in robot init.
-	 * Otherwise, it will crash since TODO: FILL-IN REASON.
+	 * Otherwise, it will crash since the OI file will try to access other subsystems once initialized.
+	 * So if all the other subsystems aren't defined prior the code will crash.
 	 */
-	public OI()
+	public OperatorInterface()
 	{
+		/**** Definitions ****/
+		
 		//Joysticks
 		controller = new Joystick(0);
 		xbox = new Joystick(1);
 		
 		//Axes
 		strafe = new Axis(controller, Robot.STRAFE);
 		
-		///Buttons
+		//Auto test buttons
+		autoTest = new JoystickButton(xbox, 3);
+		
+		//Climb buttons
+		climb = new JoystickButton(xbox, Robot.CLIMB);
+		
+		//Shoot buttons
+		shootBall = new JoystickButton(controller, Robot.SHOOT_BUTTON);
+		intakeBalls = new JoystickButton(xbox, Robot.INTAKE_BUTTON);
+
+		
+		//Vision buttons
 		center = new JoystickButton(controller, Robot.CENTER);
 		position = new JoystickButton(controller, Robot.POSITION);
 				
+		
+		//**** Links to Commands ****/
+		
 		//Drive commands
 		strafe.whileActive(new Strafe());
 		
+		//Auto test commands
+		autoTest.whenPressed(new AutoFromStation1());
+		
+		//Climb commands
+		climb.whileHeld(new Lift());
+		
+		//Shoot commands
+		shootBall.whileHeld(new ShootGroup()); //Change to ShootGroup when build adds index
+		intakeBalls.toggleWhenPressed(new IntakeBalls());
+		
 		//Vision commands
 		center.whenPressed(new CenterRobot());
 		position.whenPressed(new PositionRobot());
+		
 	}
 	
 	public double getAxis(int axis, double max)
 	{
 		return deadzone(controller.getRawAxis(axis), max);
 	}
 	
+	//Returns button value so the command group can quit when it shuts off
+	public boolean getShootButton()
+	{
+		return xbox.getRawButton(Robot.SHOOT_BUTTON);
+	}
+	
 	//TODO: Add javadoc-style comments to every function after a javadoc
```

Comment:
I'll agree with leaving the TODOs for (a little) later- we can get everything merged nicely together and then address those.